pkg/application/create.go

```diff
@@ -30,41 +33,209 @@ type CreateOp struct {
+
+		repoFullId, err := getRemoteRepositoryFullId(op, githubClient, localRepo)
+		if err != nil {
+			return result, err
+		}
+
+		pullRequest, err := createPR(op.Name, branchName, githubClient, repoFullId)
+		if err != nil {
+			return result, err
+		}
+
+		result = CreateResult{
+			MonorepoMode:       true,
+			RepositoryFullname: repoFullId.RepositoryFullname,
+			PRUrl:              pullRequest.GetHTMLURL(),
+		}
+
+	} else {
+		if err := renderTemplateMaybe(op); err != nil {
+			return result, err
+		}
+
+		if err := commitAllChanges(localRepo, "Initial commit\n[skip ci]"); err != nil {
 			return result, err
 		}
+
+		repoFullId, err := createRemoteRepository(op, githubClient, localRepo)
+		if err != nil {
+			return result, err
+		}
+		if err := synchronizeRepository(op, repoFullId, githubClient); err != nil {
+			return result, err
+		}
+
+		if err := localRepo.Push(op.GitAuth); err != nil {
+			return result, err
+		}
+
+		result = CreateResult{
+			MonorepoMode:       false,
+			RepositoryFullname: repoFullId.RepositoryFullname,
+		}
+
+	}
+	return result, nil
+}
+
+func createPR(name string, branchName string, githubClient *github.Client, repoFullId git.GithubRepoFullId) (*github.PullRequest, error) {
+	pullRequest, _, err := githubClient.PullRequests.Create(
+		context.Background(),
+		repoFullId.Organization(),
+		repoFullId.Name(),
+		&github.NewPullRequest{
+			Base:  github.String(git.MainBranch),
+			Head:  github.String(branchName),
+			Title: github.String("Add " + name + " application"),
+			Body:  github.String(fmt.Sprintf("Adding `%s` application", name)),
+		})
+	return pullRequest, err
+}
+
+func getRemoteRepositoryFullId(op CreateOp, githubClient *github.Client, localRepo *git.LocalRepository) (git.GithubRepoFullId, error) {
+	remoteRepoName, err := localRepo.GetRemoteRepoName()
+	if err != nil {
+		return git.GithubRepoFullId{}, err
+	}
+
+	githubRepo, _, err := githubClient.Repositories.Get(
+		context.Background(),
+		op.OrgName,
+		remoteRepoName,
+	)
+	if err != nil {
+		return git.GithubRepoFullId{}, err
 	}
 
+	return git.NewGithubRepoFullId(githubRepo), nil
+}
+
+func createRemoteRepository(op CreateOp, githubClient *github.Client, localRepo *git.LocalRepository) (git.GithubRepoFullId, error) {
+	githubRepo, err := createGithubRepository(op, githubClient)
+	if err != nil {
+		return git.GithubRepoFullId{}, err
+	}
+
+	repoFullId := git.NewGithubRepoFullId(githubRepo)
+
+	if err := localRepo.SetRemote(githubRepo.GetCloneURL()); err != nil {
+		return git.GithubRepoFullId{}, err
+	}
+	return repoFullId, nil
+}
+
+func undoWhenError(undoSteps *undo.Steps) {
+	if err := recover(); err != nil {
+		errs := undoSteps.Undo()
+		panic(undo.FormatError("create new application", fmt.Errorf("%v", err), errs))
+	}
+}
+
+func prepareLocalPath(localPath string, undoSteps *undo.Steps) error {
+	if err := os.MkdirAll(localPath, 0o755); err != nil {
+		return err
+	}
+	undoSteps.Add(func() error {
+		return os.RemoveAll(localPath)
```

Comment:
not sure if that's the case, but worth checking, if I create monorepo, this undo step won't delete my existing parent dir? just the submodule within that project?

Reply:
added a test